main.tf

```diff
@@ -1,14 +1,19 @@
+locals {
+  project_id        = var.project_id != null ? var.project_id : trim(substr("${var.name}-${var.env_id}", 0, 25), "-")
+  random_project_id = var.project_id == null ? true : false
+}
+
 # Provision a project for the domain
 module "domain_project_factory" {
   source  = "terraform-google-modules/project-factory/google"
   version = "~> 15.0"
 
   # Max project id is 30 chars, and we are appending a dash and 4 random chars
```

Comment:
i'd suggest moving these comments to where the local is defined

Reply:
Moved